File: Algorithm/InsortionSort.m
% ²åÈëÅÅĞò£¬Ô­Ö·
function Array=InsortionSort(Array)
len=length(Array);

for j=2:len
    key=Array(j);
    i=j-1;
    while i>0 && Array(i)>key
        Array(i+1)=Array(i);
        i=i-1;
    end
    Array(i+1)=key;
end
end

File: Algorithm/Sort & Order Statistic/C/InsortionSort.c
void InsortionSort(int* A, int arraysize)
{
	int i, j;
	int key;
	for (j = 1; j < arraysize; j++)
	{
		key = *(A + j);
		i = j - 1;
		while (i >= 0 && *(A + i)>key)
		{
			*(A + i + 1) = *(A + i);
			i--;
		}
		*(A + i + 1) = key;
	}
}